massgov/openmass
# DP-17863 private documents

**Description:**
Uploads document files to the private filesystem.


**Jira:**
https://jira.mass.gov/browse/DP-17863


**To Test:**
- [x] Go to http://mass.local/media/add/document                                    
- [x] Enter a title and upload a document to the field labeled "Select a file"        
  - Note: if you get an error stating the file cannot be uploaded this is likely because the ownership of your local private filesystem isn't correct (i.e. mine was defaulting to root:www-  data when creating the directories). To fix this run `ls -l` inside sites/default/files and check what the ownership is set to for the public directory; this should be something like `sudo  chmod -R [user-from-public-directory]:www-data`. After resetting the user, verify you can upload a document.
- [x] After uploading the document, use the link that is generated in the field to open the document in a new tab.
- [x] Verify this takes you to the private filesystem; the path should start with "system/files/documents"                                                        
- [x] Open the file URL in a private window. Verify you are unable to see the document.
- [x] After completing steps 4 and 5, save the media entity as a draft.                                      
- [x] Go to the media entity and repeat steps 4 and 5 using the `/download` link to the document on the draft.
- [x] Edit the draft media entity and set it to "Published". Save the entity.    
- [x] From the published media entity use the `/download` link to the document to verify the file is accessible from the public filesystem.
- [x] Edit the published media entity and upload a new document. Repeat the steps from 3 to 7 to ensure that during and after the editing process while the media entity is still a draft, any associated documents remain in the private filesystem.


**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)
